### PR TITLE
FLW-870 Show "missing controller" error on Redeem alert action

### PR DESCRIPTION
**Pull request by ef1rspb**

### CODE REVIEW

File: fearless/Modules/Staking/StakingMain/StakingMainPresenter.swift [hunk 1]
```diff
@@ -211,13 +245,27 @@ extension StakingMainPresenter: StakingMainPresenterProtocol {
...
+        }
         wireframe.showRedeem(from: view)
     }
 }
 
 extension StakingMainPresenter: StakingStateMachineDelegate {
-    func stateMachineDidChangeState(_: StakingStateMachineProtocol) {
+    func stateMachineDidChangeState(_ stateMachine: StakingStateMachineProtocol) {
         provideState()
+        if let stashItem = stateMachine.viewState(using: { (state: BondedState) in state.stashItem }) {
```
Comment (ERussel) [inline fearless/Modules/Staking/StakingMain/StakingMainPresenter.swift:266]: This won't work for all cases, for example when we are nominator. I would rather bind the logic when we receive stash item. Ideally, I would store stashAccount and controllerAccount in state.

Comment (ef1rspb) [reply fearless/Modules/Staking/StakingMain/StakingMainPresenter.swift:266]: oops, this is redundant code, i've removed those lines. Currently fetchController calls on stashItem
Screenshot 2021-05-20 at 14 03 08

File: fearless/Modules/Staking/StakingMain/StakingMainPresenter.swift [hunk 2]
```diff
@@ -95,16 +96,49 @@ extension StakingMainPresenter: StakingMainPresenterProtocol {
...
+                let controllerAddress = bonded.stashItem.controller
+                wireframe.presentMissingController(from: view, address: controllerAddress, locale: locale)
             }
-        } else {
-            wireframe.showSetupAmount(from: view, amount: amount)
+
+            return
+        }
+
+        let optExistingBonding: ExistingBonding? = stateMachine.viewState { (state: BondedState) in
```
Comment (ERussel) [inline fearless/Modules/Staking/StakingMain/StakingMainPresenter.swift:114]: Looks like we already have state in line 99 so we don't need to query state machine one more time

Comment (ef1rspb) [reply fearless/Modules/Staking/StakingMain/StakingMainPresenter.swift:114]: 🤦‍♂️ sure!

**Review by ERussel — APPROVED** [review]

**Pull request merged**